src/lib/syncable_ledger/syncable_ledger.ml

```diff
@@ -427,6 +458,68 @@ end = struct
     if Hash.equal actual expected then `Success
     else `Hash_mismatch (expected, actual)
 
+  (* Merges each 2 contigous nodes, halving the size of the list *)
+  let rec merge_siblings : Hash.t list -> index -> Hash.t list =
+   fun nodes height ->
+    match nodes with
+    | [ l; r ] ->
+        [ Hash.merge ~height l r ]
+    | l :: r :: rest ->
+        Hash.merge ~height l r :: merge_siblings rest height
+    | _ ->
+        (* TODO: give some error as this shouldn't really happen *)
+        []
```

Comment:
This definitely can happen if you receive 1 hash in response. Please handle this properly.